server/svix-server/src/queue/mod.rs

```diff
@@ -217,11 +247,9 @@ impl TaskQueueDelivery {
         tracing::trace!("nack {}", self.id);
         run_with_retries(
             || async {
+                // FIXME(onelson): borrowing &self in the match allows redis arm to pass it to `nack`
+                //   Omni arm needs `mut self` for `delivery.nack()`. How to satisfy both?
```

Comment:
I think this was the active concern when I was looking at this last.

The plan was to redo omniqueue to use interior mutability so the Acker in there can be `&self` (non-mut) like the rest.

The mutability in omni was needed for a "ack or nack once" check, and it looks like we could use `AtomicBool` for this and sidestep the issue.